messaging/artemis-jta/src/main/java/io/quarkus/ts/openshift/messaging/artemisjta/ConsumerService.java

```diff
@@ -38,28 +42,14 @@ public void setPrice2(String price) {
         this.price2 = price;
     }
 
-    @Scheduled(every = "1s")
-    public void receiveMessages() throws JMSException {
-        try (JMSContext context = connectionFactory.createContext(Session.AUTO_ACKNOWLEDGE)) {
-            JMSConsumer consumer = context.createConsumer(context.createQueue("custom-prices-1"));
-            Message message = consumer.receive(500);
-            if (message != null) {
-                String price = message.getBody(String.class);
-                setPrice1(price);
-                LOG.info("Price 1 set to " + price);
-            } else {
-                LOG.info("Nothing to see in queue custom-prices-1.");
-            }
-            consumer = context.createConsumer(context.createQueue("custom-prices-2"));
-            message = consumer.receive(500);
-            if (message != null) {
-                String price = message.getBody(String.class);
-                setPrice2(price);
-                LOG.info("Price 2 set to " + price);
-            } else {
-                LOG.info("Nothing to see in queue custom-prices-2.");
-            }
-        }
+    @Scheduled(every = "1s", concurrentExecution = ConcurrentExecution.SKIP)
+    public void receiveMessagesInCustomPriceOne() throws JMSException {
+        receiveMessagesInQueue("custom-prices-1", this::setPrice1);
+    }
+
+    @Scheduled(every = "1s", concurrentExecution = ConcurrentExecution.SKIP)
+    public void receiveMessagesInCustomPriceTwo() throws JMSException {
+        receiveMessagesInQueue("custom-prices-2", this::setPrice2);
```

Comment:
I made this change to fix the issue in this comment: https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/106#issuecomment-720936901

With concurrent configured to skip, we disallow to trigger the same job if there is a running previous one. Moreover, we avoid to ack messages from two queues at the same time (in receiveMessagesInQueue).